pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -71,14 +71,37 @@ type InfinispanSiteLocationSpec struct {
 }
 
 type InfinispanSitesSpec struct {
-	Local   InfinispanSitesLocalSpec    `json:"local"`
+	Local     InfinispanSitesLocalSpec     `json:"local"`
 	Locations []InfinispanSiteLocationSpec `json:"locations"`
 }
 
 type InfinispanLoggingSpec struct {
 	Categories map[string]string `json:"categories"`
 }
 
+// ExposeType describe different exposition methods for Infinispan
+type ExposeType string
+
+const (
+	// ExposeTypeNodePort means a service will be exposed on one port of
+	// every node, in addition to 'ClusterIP' type.
+	ExposeTypeNodePort ExposeType = "NodePort"
+
+	// ExposeTypeLoadBalancer means a service will be exposed via an
+	// external load balancer (if the cloud provider supports it), in addition
+	// to 'NodePort' type.
+	ExposeTypeLoadBalancer ExposeType = "LoadBalancer"
+
+	// ExposeTypeRoute means Infinispan cluster will be exposed via Route resource
+	ExposeTypeRoute ExposeType = "Route"
```

Comment:
This new option doesn't seem to be handled later in the code.

Reply:
I've tried to add support for Route exposition, but openshift packages conflict with the current dependencies set, so I've posponed this feature to 2.x.
It's probably better to remove all the references to Route for now. Going to do.